### ISSUE TITLE
Reset Password will not work when email address contains capital letter.

### ISSUE THREAD

**Issue by tkforbes** [RocketChat/Rocket.Chat#28730]
This reference page does not exist ... https://rocket.chat/docs/contributing/reporting-issues

### Description:

If the email address includes a capital letter (upper case letter), the "Reset Password" feature is not triggered. The Reset Password feature is accessed from the Forgot Password link on the login page

### Steps to reproduce:

1. Go to Login.
2. Click "Forgot your password".
3. Enter email address on Reset Password page. Press [Send instructions] button.
4. No email is sent.

### Expected behavior:

1. Go to Login.
2. Click "Forgot your password".
3. Enter email address on Reset Password page. Press [Send instructions] button.
4. Receive an email at registered address.

### How I proved this.

Monitored the logs of my email server. No email arrived at the server.
Using admin account in RC, updated the users email address several times.
Only when the capital letter was dropped would an email actually be sent.

Note:

There is an existing bug in RC where the case of an email address cannot be changed. I presume RC thinks the addresses are equivalent regardless of case. See...

https://github.com/RocketChat/Rocket.Chat/issues/12557

**Comment by Dev79844** [comment]
Hey
I would like to work on this issue. Can you assign it to me?

**Comment by tkforbes** [comment]
Hi Dev. I was notified about your comment. I checked if I could assign the issue, but I'm unable. I'm assuming the assignment would need to be done by the repo owner.

**Comment by 01vishu** [comment]
I am interested in contributing to your project and would like to request to be assigned to the project. I believe I have the necessary skills and expertise to make meaningful contributions to the project.

**Comment by Pranshu1902** [comment]
@tkforbes I would like to work on this issue. Do I need to wait until I get assigned to this issue or can I raise a PR straight away?

**Comment by tkforbes** [comment]
I reported the issue but I think it is up to someone associated with the project to assign the work. I'm unable to do anything but comment here.

**Comment by hugocostadev** [comment]
Anyone can work in any issue and reference this issue in the PR, no need to ask permission

**Comment by adarsh-sgh** [comment]
Rocket Chat is assuming emails to be case-insensitive.
In practice [no widely used mail systems distinguish](https://stackoverflow.com/questions/9807909/are-email-addresses-case-sensitive) different addresses based on case.
such assumption gives the advantage that users don't need to care about case of the email address while typing.

I can try to change it to make rocket chat emails case sensitive but I think it will be best to keep it as a setting option and disabling case sensitivity in emails by default as it is currently🤔 
any suggestions ?

**Comment by tkforbes** [comment]
My opinion:

At the point that a user needs to reset their password, whatever Rocket Chat has recorded as their email address must be considered valid. 

When I ran across this issue, I was setting up a test RC instance and defining multiple accounts for my own use. I defined a user whose email address included an upper case letter without rejection. It was therefore bewildering that I could not reset the password of that user considering that the email was (rightfully) accepted as valid at the time it was defined.

Regarding the comment in the previous post about Adarsh@example.com going to adarsh@example.com, I did not observe that behaviour.  I happen to be running the email servers of both my RC domain and my personal domain and the log files of both of those servers were quiet when I attempted to reset the password.
